File: app/src/main/java/com/kota/ASFramework/PageController/ASViewRemover.java
package com.kota.ASFramework.PageController;

import android.view.View;
import android.view.ViewGroup;
import com.kota.ASFramework.Thread.ASRunner;

/* loaded from: classes.dex */
public class ASViewRemover {
  private final ViewGroup _parent_view;
  private final View _target_view;

  public static void remove(ViewGroup parentView, View targetView) {
    ASViewRemover remover = new ASViewRemover(parentView, targetView);
    remover.start();
  }

  public ASViewRemover(ViewGroup parentView, View targetView) {
    this._parent_view = parentView;
    this._target_view = targetView;
  }

  public void start() {
    ASRunner.runInNewThread(()->{
        try {
          Thread.sleep(1000L);
        } catch (InterruptedException e) {
          e.printStackTrace();
        }
        ASViewRemover.this.remove();
    });
  }

  private void remove() {
    new ASRunner() { // from class: com.kota.ASFramework.PageController.ASViewRemover.2
      @Override // com.kota.ASFramework.Thread.ASRunner
      public void run() {
        if (ASViewRemover.this._parent_view != null && ASViewRemover.this._target_view != null) {
          ASViewRemover.this._parent_view.removeView(ASViewRemover.this._target_view);
        }
      }
    }.runInMainThread();
  }
}

File: app/src/main/java/com/kota/asFramework/pageController/ASViewRemover.kt
package com.kota.asFramework.pageController

import android.view.View
import android.view.ViewGroup
import com.kota.asFramework.thread.ASCoroutine

class ASViewRemover(private val parentViewGroup: ViewGroup?, private val targetView: View?) {
    fun start() {
        ASCoroutine.runInNewCoroutine {
            try {
                Thread.sleep(1000L)
            } catch (e: InterruptedException) {
                e.printStackTrace()
            }
            this@ASViewRemover.remove()
        }
    }

    private fun remove() {
        ASCoroutine.ensureMainThread {
            if (this@ASViewRemover.parentViewGroup != null && this@ASViewRemover.targetView != null) {
                this@ASViewRemover.parentViewGroup.removeView(this@ASViewRemover.targetView)
            }
        }
    }

    companion object {
        fun remove(parentView: ViewGroup?, targetView: View?) {
            val remover = ASViewRemover(parentView, targetView)
            remover.start()
        }
    }
}